src/compiler/program.ts

```diff
@@ -26,6 +26,11 @@ module ts {
         return undefined;
     }
 
+    /**
+     * @param setParentNodes Whether or not parent references should be set on nodes.
+     *                       This is useful when one desires parent references, but the semantic information of binding is unnecessary.
+     */
+    export function createCompilerHost(options: CompilerOptions, setParentNodes = true): CompilerHost {

```

Comment:
if you make it true, you need to make all places that use it today explicitly false.